src/cltkv1/utils/data_types.py

```diff
@@ -73,6 +72,29 @@ class Word:
     feats: str = None  # morphological features (from stanfordnlp)
 
 
+@dataclass
+class Doc:
+    """The object returned to the user from the ``NLP()`` class.
+    Contains overall attributes of submitted texts, plus most
+    importantly the processed tokenized text ``words``,
+    being a list of ``Word`` types.
+    """
+
+    indices_sentences: List[List[int]] = None
+    indices_tokens: List[List[int]] = None
+    language: str = None
+    words: List[Word] = None
+    pipeline: List["Process"] = None
+    raw: str = None
+
+    @property
+    def sentences(self):
```

Comment:
You have anticipated what I was imaging -- a variety of access methods/properties to pull what users will conventionally want. So `Doc.sentences` and (say) `Doc. dependency_tree` and `Doc.pos` etc. These methods would usually read from `Doc.words` and return a summarization of this info.

Are we thinking similarly here? Seem like the most comfortable interface for the average user?

Reply:
Yes exactly.  `Doc.words` plus whatever extra info is needed (like the token indices for sentences) are read by useful accessors.